app/toolbox.cpp

```diff
@@ -35,130 +36,102 @@ QString GetToolTips(QString strCommandName)
-                                .arg(GetToolTips(CMD_TOOL_SELECT)));
-    mMoveButton->setWhatsThis(tr("Move Tool (%1)")
-                              .arg(GetToolTips(CMD_TOOL_MOVE)));
-    mHandButton->setWhatsThis(tr("Hand Tool (%1)")
-                              .arg(GetToolTips(CMD_TOOL_HAND)));
-    mPenButton->setWhatsThis(tr("Pen Tool (%1)")
-                             .arg(GetToolTips(CMD_TOOL_PEN)));
-    mEraserButton->setWhatsThis(tr("Eraser Tool (%1)")
-                                .arg(GetToolTips(CMD_TOOL_ERASER)));
-    mPolylineButton->setWhatsThis(tr("Polyline Tool (%1)")
-                                  .arg(GetToolTips(CMD_TOOL_POLYLINE)));
-    mBucketButton->setWhatsThis(tr("Paint Bucket Tool (%1)")
-                                .arg(GetToolTips(CMD_TOOL_BUCKET)));
-    mBrushButton->setWhatsThis(tr("Brush Tool (%1)")
-                               .arg(GetToolTips(CMD_TOOL_BRUSH)));
-    mEyedropperButton->setWhatsThis(tr("Eyedropper Tool (%1)")
-                                    .arg(GetToolTips(CMD_TOOL_EYEDROPPER)));
-    mClearButton->setWhatsThis(tr("Clear Tool (%1)")
-                               .arg(GetToolTips(CMD_CLEAR_FRAME)));
-    mSmudgeButton->setWhatsThis(tr("Smudge Tool (%1)")
-                                .arg(GetToolTips(CMD_TOOL_SMUDGE)));
-
-    mPencilButton->setCheckable(true);
-    mPenButton->setCheckable(true);
-    mPolylineButton->setCheckable(true);
-    mBucketButton->setCheckable(true);
-    mBrushButton->setCheckable(true);
-    mSmudgeButton->setCheckable(true);
-    mEyedropperButton->setCheckable(true);
-    mSelectButton->setCheckable(true);
-    mMoveButton->setCheckable(true);
-    mHandButton->setCheckable(true);
-    mEraserButton->setCheckable(true);
-    mPencilButton->setChecked(true);
-
-    mGridLayout->setMargin(0);
-    mGridLayout->setSpacing(0);
-
-    QWidget* toolGroup = new QWidget(this);
-
-    setWidget(toolGroup);
-    toolGroup->setLayout(mGridLayout);
-
-    connect(mPencilButton, &QToolButton::clicked, this, &ToolBoxWidget::pencilOn);
-    connect(mEraserButton, &QToolButton::clicked, this, &ToolBoxWidget::eraserOn);
-    connect(mSelectButton, &QToolButton::clicked, this, &ToolBoxWidget::selectOn);
-    connect(mMoveButton, &QToolButton::clicked, this, &ToolBoxWidget::moveOn);
-    connect(mPenButton, &QToolButton::clicked, this, &ToolBoxWidget::penOn);
-    connect(mHandButton, &QToolButton::clicked, this, &ToolBoxWidget::handOn);
-    connect(mPolylineButton, &QToolButton::clicked, this, &ToolBoxWidget::polylineOn);
-    connect(mBucketButton, &QToolButton::clicked, this, &ToolBoxWidget::bucketOn);
-    connect(mEyedropperButton, &QToolButton::clicked, this, &ToolBoxWidget::eyedropperOn);
-    connect(mBrushButton, &QToolButton::clicked, this, &ToolBoxWidget::brushOn);
-    connect(mSmudgeButton, &QToolButton::clicked, this, &ToolBoxWidget::smudgeOn);
-    connect(mClearButton, &QToolButton::clicked, this, &ToolBoxWidget::clearButtonClicked);
```

Comment:
I won't recommend doing connections with ui files. UI file is good at positioning GUI things, but it could be annoying that it deals with logic as well. For example, if I want to know the code flow of `pencilOn()` function and search the function name in the entire project, I will finally find it in an XML file like this 
```xml
<connection>
   <sender>pencilButton</sender>
   <signal>clicked()</signal>
   <receiver>ToolBoxWidget</receiver>
   <slot>pencilOn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>381</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>99</y>
    </hint>
   </hints>
</connection>
```
The readability is quite bad comparing to a single line connection in a .cpp file
```cpp
connect(mPencilButton, &QToolButton::clicked, this, &ToolBoxWidget::pencilOn);
```

Also I couldn't see any benefits to do connections in ui files.

Reply:
Ah yes my reason for this was separation but readability is certainly an important aspect to keep track of. Probably going to revert that then, but I’m going to think about it properly tomorrow when I’m not tired.

Reply:
Thought about it some more, and it seems that in regard to readability there really is no benefit in keeping the connections in the UI file even know that it’s already done, so I’m going to move the connections back into the code once I find some time for it.